backend/tool_instance_v2/serializers.py

```
@@ -51,8 +55,102 @@ def to_representation(self, instance: ToolInstance) -> dict[str, str]:
+                            logger.error(
+                                f"Could not resolve adapter ID {adapter_value} to name: {e}"
+                            )
+                    # If it's not a UUID, validate if the adapter name still exists
+                    elif adapter_value and not ToolInstanceHelper.is_uuid_format(
+                        adapter_value
+                    ):
+                        try:
+                            # Validate if adapter name still exists
+                            adapter = AdapterProcessor.get_adapter_by_name_and_type(
+                                adapter_type=adapter_type, adapter_name=adapter_value
+                            )
+                            if adapter:
+                                # Adapter name is valid, display as-is
+                                display_metadata[adapter_key] = adapter_value
+                            else:
+                                # Adapter name no longer exists (likely renamed)
+                                display_metadata[adapter_key] = (
+                                    f"{adapter_value} [NEEDS UPDATE]"
+                                )
+                                logger.warning(
+                                    f"Adapter name '{adapter_value}' no longer exists for type {adapter_type} - may have been renamed"
+                                )
+                        except Exception as e:
+                            # If validation fails, show error
+                            display_metadata[adapter_key] = f"[{adapter_value} NOT FOUND]"
+                            logger.error(
+                                f"Could not validate adapter name '{adapter_value}': {e}"
+                            )
+
+        except Exception as e:
+            logger.warning(f"Error transforming adapter IDs to names: {e}")
+            # Return original metadata if transformation fails
+            pass
+
+        return display_metadata
```

@kirtimanmishrazipstack in case of an exception, we'll be failing silently and return the metadata with UUIDs instead right? Won't it break elsewhere / still be unusable for users? It might be good to just raise the exception and then fix it